test/integration-tests/IntegrationTests.Http/HttpTests.cs

```diff
@@ -0,0 +1,82 @@
+// <copyright file="HttpTests.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System.Linq;
+using FluentAssertions.Execution;
+using IntegrationTests.Helpers;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace IntegrationTests.Http
+{
+    [Collection(HttpCollection.Name)]
+    public class HttpTests : TestHelper
+    {
+        private const string ServiceName = "Samples.Http";
+
+        public HttpTests(ITestOutputHelper output)
+            : base("Http", output)
+        {
+            SetEnvironmentVariable("OTEL_SERVICE_NAME", ServiceName);
+            SetEnvironmentVariable("OTEL_DOTNET_AUTO_ENABLED_INSTRUMENTATIONS", "HttpClient,AspNet");
+        }
+
+        [Fact]
+        [Trait("Category", "EndToEnd")]
+        public void SubmitTraces()
+        {
+            var agentPort = TcpPortProvider.GetOpenPort();
+            using var agent = new MockZipkinCollector(Output, agentPort);
+
+            const int expectedSpanCount = 3;
+
+            using var processResult = RunSampleAndWaitForExit(agent.Port, enableStartupHook: true);
+            Assert.True(processResult.ExitCode >= 0, $"Process exited with code {processResult.ExitCode} and exception: {processResult.StandardError}");
+            var spans = agent.WaitForSpans(expectedSpanCount, 500);
+
+            using var scope = new AssertionScope();
+            Assert.True(spans.Count == expectedSpanCount, $"Expecting {expectedSpanCount} spans, received {spans.Count}");
+
+            var httpClientSpan = spans.FirstOrDefault(span => span.Name.Equals("HTTP GET"));
+            var httpServerSpan = spans.FirstOrDefault(span => span.Name.Equals("/test"));
+            var manualSpan = spans.FirstOrDefault(span => span.Name.Equals("manual span"));
+
+            Assert.NotNull(httpClientSpan);
+            Assert.NotNull(httpServerSpan);
```

Comment:
maybe it would be worth adding comments like `// ASP.NET Core auto-instrumentation is generating spans` (same comment applies to other places where it seems fit for your)